public/Connect-DbaInstance.ps1

```diff
@@ -645,15 +645,39 @@ function Connect-DbaInstance {
             } elseif ($inputObjectType -eq 'SqlConnection') {
                 $server = New-Object -TypeName Microsoft.SqlServer.Management.Smo.Server -ArgumentList $inputObject
             } elseif ($inputObjectType -in 'RegisteredServer', 'ConnectionString') {
-                $server = New-Object -TypeName Microsoft.SqlServer.Management.Smo.Server -ArgumentList $serverName
-                # Parameter TrustServerCertificate changes the connection string be allow connections to instances with the default self-signed certificate
+                # Create the server SMO in the same way as when passing a string (see #8962 for details).
+                # Best way to get connection pooling to work is to use SqlConnectionInfo -> ServerConnection -> Server
+                $sqlConnectionInfo = New-Object -TypeName Microsoft.SqlServer.Management.Common.SqlConnectionInfo
+
+                # Set properties of SqlConnectionInfo based on the used properties of the connection string.
+                $csb = New-Object -TypeName Microsoft.Data.SqlClient.SqlConnectionStringBuilder -ArgumentList $connectionString
```

Comment:
To pass arguments using new object creation in PS 5+ example:

```suggestion
                $csb = [Microsoft.Data.SqlClient.SqlConnectionStringBuilder]::new($connectionString)
```

Just examples no change required at this point.

Reply:
Wait no way, we cant use ::new, that is not supported in PS3